inference-engine/tests/functional/plugin/shared/src/single_layer_tests/fake_quantize.cpp

```diff
@@ -18,47 +18,133 @@
+    std::tie(fqParams, netPrecision, inputShape, targetDevice, config) = this->GetParam();
     InferenceEngine::SizeVector kernel, stride, dilation;
     size_t levels;
     std::vector<size_t> constShape;
-    std::tie(levels, constShape) = fqParams;
+    std::vector<float> fqDirectArg;
+    std::vector<float> inputArg;
+    std::tie(levels, constShape, fqDirectArg, inputArg) = fqParams;
+    if (inputArg.size() == 3) {
+        inputDataMin = inputArg[0];
+        inputDataMax = inputArg[1];
+        inputDataResolution = inputArg[2];
+    }
     auto ngPrc = FuncTestUtils::PrecisionUtils::convertIE2nGraphPrc(netPrecision);
     auto params = ngraph::builder::makeParams(ngPrc, {inputShape});
     auto paramOuts = ngraph::helpers::convert2OutputVector(ngraph::helpers::castOps2Nodes<ngraph::op::Parameter>(params));
 
-    auto fq = std::dynamic_pointer_cast<ngraph::opset1::FakeQuantize>(ngraph::builder::makeFakeQuantize(paramOuts[0], ngPrc, levels, constShape));
+    UpdateSeed();
+
+    std::shared_ptr<ngraph::Node> fakeQNode;
+    if (fqDirectArg.empty()) {
+        int32_t ngraphSeed = seed;
+        if (NGRAPH_SEED != USE_CLOCK_TIME) {
+            ngraphSeed = NGRAPH_SEED;
+        }
+        std::cout << "\033[0;32m" << "[          ] " << "\033[0;0m"
+                  << "ngraphSeed = " << ngraphSeed << std::endl;
+        fakeQNode = ngraph::builder::makeFakeQuantize(paramOuts[0], ngPrc, levels, constShape, ngraphSeed);
+    } else {
+        fakeQNode = ngraph::builder::makeFakeQuantize(
+            paramOuts[0],
+            ngPrc,
+            levels,
+            constShape,
+            {fqDirectArg[0]},
+            {fqDirectArg[1]},
+            {fqDirectArg[2]},
+            {fqDirectArg[3]});
+    }
+
+
+    auto fq = std::dynamic_pointer_cast<ngraph::opset1::FakeQuantize>(fakeQNode);
 
     ngraph::ResultVector results{std::make_shared<ngraph::opset1::Result>(fq)};
     function = std::make_shared<ngraph::Function>(results, params, "fakeQuantize");
+
+    configuration = config.second;
+}
+
+InferenceEngine::Blob::Ptr FakeQuantizeLayerTest::GenerateInput(const InferenceEngine::InputInfo &info) const {
+    return FuncTestUtils::createAndFillBlob(info.getTensorDesc(), inputDataMax - inputDataMin, inputDataMin, 1 / inputDataResolution, seed);
+}
+
+void FakeQuantizeLayerTest::UpdateSeed() {
+    if (BASE_SEED == USE_CLOCK_TIME) {
+        seed = std::chrono::system_clock::now().time_since_epoch().count();
+    } else if (BASE_SEED == USE_INCREMENTAL_SEED) {
+        seed += 9999;
+    } else {
+        seed = BASE_SEED;
+    }
+    std::cout << "\033[0;32m" << "[          ] " << "\033[0;0m"
+              << "seed = " << seed << std::endl;
 }
 
 TEST_P(FakeQuantizeLayerTest, CompareWithRefs) {
     Run();
+    SKIP_IF_CURRENT_TEST_IS_DISABLED();
+
+    if (BASE_SEED != USE_CLOCK_TIME &&
+        BASE_SEED != USE_INCREMENTAL_SEED) {
+        return;
+    }
```

Comment:
Nevermind, i've  noticed that there is Run() above, but why SKIP_IF_CURRENT_TEST_IS_DISABLED() is below Run()?

Reply:
ok long story short - in base class's run() skip can be triggered as well. I would not modify base check since it might do something before this check inside run() , so decided to call it after base. In fact it doesn't mater much. But who knows one day check in base Run might be altered. Right not CPU tests are disabled that is why i added check here as well.